revoke.go

```diff
@@ -31,18 +32,18 @@ func revocationEndpoint(p *oidc.Provider) (string, error) {
 // revokeTokens is a helper that takes an oauth2.Token and revokes the access and refresh tokens.
 // If no tokens are found, it succeeds.
 func revokeTokens(ctx context.Context, revocationEndpoint string, token *oauth2.Token, clientID, clientSecret string) error {
-	if token.AccessToken != "" {
-		err := revokeToken(ctx, revocationEndpoint, token.AccessToken, "access_token", clientID, clientSecret)
-		if err != nil {
-			return errors.Wrap(err, "Failed to revoke access token")
-		}
-	}
 	if token.RefreshToken != "" {
 		err := revokeToken(ctx, revocationEndpoint, token.RefreshToken, "refresh_token", clientID, clientSecret)
 		if err != nil {
 			return errors.Wrap(err, "Failed to revoke refresh token")
 		}
 	}
+	if token.AccessToken != "" {
+		err := revokeToken(ctx, revocationEndpoint, token.AccessToken, "access_token", clientID, clientSecret)
+		if err != nil {
+			log.Warning("Failed to revoke access token")
```

Comment:
We shouldn't just catch all errors and throw a warning. RFC 7009 has a section on [error responses](https://datatracker.ietf.org/doc/html/rfc7009#section-2.2.1):

   > unsupported_token_type:  The authorization server does not support the revocation of the presented
   > token type. That is, the client tried to revoke an access token on a server not supporting this feature.
   
So, we should catch HTTP 400 errors and check for the `unsupported_token_type` error code. The rest of the error responses are valid errors and we should return them as we already do.